minio/minio
# fix: lifecycle-expiration validation bug

## Description
Currently, when you set a lifecycle  to a bucket:
1. you can set Expiration with both Days and Date, only Date works.
![image](https://user-images.githubusercontent.com/17903517/90961129-4a12be80-e4d9-11ea-8eb8-ecfab439ce54.png)

2. when you set Days=0, the error message is confusing.
![image](https://user-images.githubusercontent.com/17903517/90961136-5a2a9e00-e4d9-11ea-872a-4221b2161dac.png)

It seems that:
1. unit-tests covered the above two cases in `expiration_test.go.go`, but not in`lifecycle_test.go`.
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/pkg/bucket/lifecycle/expiration_test.go#L26
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/pkg/bucket/lifecycle/lifecycle_test.go#L27

2. cases covered in xml-Unmarshal not working when you do http-body decode
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/pkg/bucket/lifecycle/expiration.go#L43-L45
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/cmd/bucket-lifecycle-handlers.go#L69


## Motivation and Context
improve user experience

## How to test this PR?
set a lifecycle  to a bucket:
1. set Expiration with both Days and Date, and check the error message
2. set Expiration Days=0, and check the error message


## Types of changes

## Checklist:
